File: app/src/main/java/com/alexis/funchat/base/BaseFragment.java
package com.alexis.funchat.base;

import android.os.Bundle;

import com.alibaba.android.arouter.launcher.ARouter;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;


/**
 * Class description:
 *
 * @author Alexis
 * @date 2019/5/14  10:26
 */
public abstract class BaseFragment extends Fragment {
    protected ARouter mRouter;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mRouter = ARouter.getInstance();

    }

    public void navigation(String path){
        mRouter.build(path).navigation();
    }
}

File: app/src/main/java/com/alexis/funchat/base/BaseFragment.kt
package com.alexis.funchat.base

import android.os.Bundle

import com.alibaba.android.arouter.launcher.ARouter
import androidx.fragment.app.Fragment


/**
 * Class description:
 *
 * @author Alexis
 * @date 2019/5/14  10:26
 */
abstract class BaseFragment : Fragment() {
    protected lateinit var mRouter: ARouter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mRouter = ARouter.getInstance()

    }

    fun navigation(path: String) {
        mRouter.build(path).navigation()
    }
}
